File: src/app/material-advisor/src/app/features/topic/knowledge-check/models/KnowledgeCheck.ts
import { GUID } from "@shared/types/GUID";

export interface KnowledgeCheck {
  id: GUID | null;
  topicId: GUID | null;
  name: string;
  description: string;
  startDate: Date;
  endDate: Date;
  time: number;
  maxAttempts: number;
  groupIds: GUID[];
}

File: src/api/MaterialAdvisor/MaterialAdvisor.Application/Models/KnowledgeChecks/KnowledgeCheck.cs
namespace MaterialAdvisor.Application.Models.KnowledgeChecks;

public class KnowledgeCheck
{
    public Guid Id { get; set; }

    public Guid TopicId { get; set; }

    public string Name { get; set; } = null!;

    public string Description { get; set; } = null!;

    public DateTime StartDate { get; set; }

    public DateTime? EndDate { get; set; }

    public ushort? Time { get; set; }

    public byte? MaxAttempts { get; set; }

    public byte? UsedAttempts { get; set; }

    public IEnumerable<Guid> GroupIds { get; set; } = [];
}
